source/dofs/dof_tools_constraints.cc

```diff
@@ -1584,12 +1660,152 @@ namespace DoFTools
 
                       case FiniteElementDomination::neither_element_dominates:
                       {
-                        // we don't presently know what exactly to do here.
-                        // it isn't quite clear what exactly we would have
-                        // to do here. sit tight until someone trips over
-                        // the following statement and see what exactly is
-                        // going on
-                        Assert (false, ExcNotImplemented());
+                        // make sure we don't get here twice from each cell
+                        if (processed_p_faces.find(cell->face(face)) != processed_p_faces.end())
+                          break;
```

Comment:
i don't think so. These non-refined faces, and there is a loop over cells and then over it's faces. So we definitely visit each face twice. This comment for the case when `other_element_dominates ` confirms it:
```
case FiniteElementDomination::other_element_dominates:
                      {
                        // we don't do anything here since we will come
                        // back to this face from the other cell, at which
                        // time we will fall into the first case clause
                        // above
                        break;
                      }
```